python_ta/checkers/unnecessary_indexing_checker.py

```diff
@@ -106,8 +128,14 @@ def _is_redundant(index_node: Union[astroid.AssignName, astroid.Name], for_node:
     in the for loop's body.
     """
     if isinstance(index_node, astroid.AssignName):
-        return index_node.lookup(index_node.name)[1][0] != for_node.target or \
-               not isinstance(index_node.parent, astroid.AugAssign)
```

Comment:
I'm still confused about this condition for AugAssign, because I don't really understand what the checker is supposed to flag that doesn't involve directly subscripting the iterable (based on the documentation). If I don't change this then some of the cases based off the example files (nested comprehensions (2, 3, 4), `test_loop_variable_reassigned_no_msg`, `test_comp_shadow_no_msg`) will fail. In the documentation for this checker it does say that it may not work properly for cases like these and raise a false error, so I'm wondering if these cases are supposed to be ignored?

Reply:
@lorena-b I'm not sure exactly what you're asking. The tests you have added look good, and no cases should be ignored. In fact, if you have improved the checker, then you should update the documentation to remove this comment. It is okay to modify the code in this branch (after all, there was a bug), and my previous comment was that you should not remove the branch entirely.

However, there is a case that is suggested by the `AugAssign` that you haven't tested, and that's when we have a statement like `lst[i] += 1`. In this case, even though the `i` is inside a subscript node, it is not redundant. You should add a test for this case as well, and ensure that this implementation correctly handles it.